src/tests/ftest/util/network_utils.py

```diff
@@ -0,0 +1,560 @@
+
+
+def get_active_network_interfaces(hosts, verbose=True):
+    """Get all active network interfaces on the hosts.
+
+    Args:
+        hosts (NodeSet): hosts on which to find active interfaces
+        verbose (bool, optional): display command details. Defaults to True.
+
+    Returns:
+        dict: a dictionary of interface keys and NodeSet values on which they were found
+
+    """
+    net_path = os.path.join(os.path.sep, "sys", "class", "net")
+    operstate = os.path.join(net_path, "*", "operstate")
+    command = " | ".join(
+        ["grep -l 'up' {}".format(operstate), "grep -Ev '/(lo|bonding_masters)/'", "sort"])
+    task = run_task(hosts, command, verbose=verbose)
+    if verbose:
+        display_task(task)
+
+    # Populate a dictionary of active interfaces with a NodSet of hosts on which it was found
+    active_interfaces = {}
+    for output, nodelist in task.iter_buffers():
+        output_lines = [line.decode("utf-8") for line in output]
+        nodeset = NodeSet.fromlist(nodelist)
+        for line in output_lines:
+            try:
+                interface = line.split("/")[-2]
+                if interface not in active_interfaces:
+                    active_interfaces[interface] = NodeSet()
+                active_interfaces[interface].update(nodeset)
+            except IndexError:
+                pass
+
+    return active_interfaces
+
+
+def get_common_interfaces(hosts, verbose=True):
+    """Get the active interfaces that exists on all the hosts.
+
+    Args:
+        hosts (NodeSet): hosts on which to find active interfaces
+        verbose (bool, optional): display command details. Defaults to True.
+
+    Returns:
+        list: a list of available NetworkDevice objects common to all hosts
+
+    """
+    log = getLogger()
+
+    # Find any active network interfaces on the server or client hosts
+    interfaces_per_host = get_active_network_interfaces(hosts, verbose)
+
+    # From the active interface dictionary find all the interfaces that are common to all hosts
+    log.info("Active network interfaces detected:")
+    common_interfaces = []
+    for interface, node_set in interfaces_per_host.items():
+        log.info("  - %-8s on %s (Common=%s)", interface, node_set, node_set == hosts)
+        if node_set == hosts:
+            common_interfaces.append(interface)
+
+    return common_interfaces
+
+
+def get_interface_speeds(hosts, interface, verbose=True):
+    """Get the speeds of this network interface on each host.
+
+    Args:
+        hosts (NodeSet): hosts on which to detect the interface speed
+        interface (str): interface for which to obtain the speed
+        verbose (bool, optional): display command details. Defaults to True.
+
+    Returns:
+        dict: a dictionary of interface speed keys and NodeSet values on which they were detected
+
+    """
+    net_path = os.path.join(os.path.sep, "sys", "class", "net")
+    command = "cat {}".format(os.path.join(net_path, interface, "speed"))
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string